### PR TITLE
`SurfaceKinetics` BC

### PR DESCRIPTION
**Pull request by KulaginVladimir**
## Proposed changes

This PR represents the initial attempt on implementing Surface Kinetics feature as a new boundary condition in FESTIM and is aimed at starting the discussion. Additionally, a specific `DerivedQuantity` is introduced to export the surface concentration of H.

To-do list:
- [ ] Tests
- [ ] Docs (API, UG, Theory)
- [x] Error message when the `SurfaceKinetics` BC is set with other BCs on the same surface
- [x] Error message when the `SurfaceKinetics` BC is used not in 1D simulations
- [x] Error message when the `AdsorbedHydrogen` is called without setting `SurfaceKinetics` on this surface (@RemDelaporteMathurin, is this needed?)

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: festim/concentration/mobile.py [hunk 1]
```diff
@@ -184,12 +184,25 @@ def create_fluxes_form(self, T, ds):
...
+                        bc.create_form(
+                            solute,
+                            self.previous_solution,
+                            self.test_function,
+                            T,
+                            ds,
+                            dt,
+                        )
+                        F += bc.F
+                        expressions_fluxes += bc.sub_expressions
```
Comment (RemDelaporteMathurin) [inline festim/concentration/mobile.py:197]: don't think this line is needed since it's already added on line 204

File: festim/exports/derived_quantities/adsorbed_hydrogen.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @property
+    def title(self):
+        quantity_title = f"Concentration of adsorbed H on surface {self.surface}"
+        if self.show_units:
+            return quantity_title + f" ({self.export_unit})"
+        else:
+            return quantity_title
+
+    def compute(self):
+        return f.assemble(self.function * self.ds(self.surface))
```
Comment (RemDelaporteMathurin) [inline festim/exports/derived_quantities/adsorbed_hydrogen.py:39]: If this class is used for a non-0D surface then the units will not be H/m2. 
This is fine because this functionality uses Real elements anyway, but I would add assert statements in a few places to make sure this is only used for 1D simulations

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Hi @KulaginVladimir thanks so much for this! I think a lot of people will use this new functionality! (ping @ehodille )
Here's a first round of comments on my side

File: festim/h_transport_problem.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from fenics import *
 import festim
 import warnings
+import sympy as sp
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/h_transport_problem.py:4]: `warnings` as well

File: festim/exports/derived_quantities/adsorbed_hydrogen.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+from festim import SurfaceQuantity, k_B
+import fenics as f
+
+
+class AdsorbedHydrogen(SurfaceQuantity):
+    """
+    Computes the hydrogen surface concentration on a surface
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/exports/derived_quantities/adsorbed_hydrogen.py:7]: Do you mean the export can be done with another class or the name/description should be changed?

File: festim/h_transport_problem.py [hunk 5]
```diff
@@ -141,11 +159,31 @@ def initialise_concentrations(self):
...
+                    # iterate through each surface of each SurfaceKinetics
+                    for j in range(len(concentration.surfaces)):
+                        concentration.solutions[j] = self.u.sub(i + j)
+                        concentration.previous_solutions[j] = self.u_n.sub(i + j)
+                        concentration.test_functions[j] = list(split(self.v))[i + j]
+                else:
+                    concentration.solution = self.u.sub(i)
+                    # concentration.solution = list(split(self.u))[i]
+                    concentration.previous_solution = self.u_n.sub(i)
+                    concentration.test_function = list(split(self.v))[i]
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/h_transport_problem.py:186]: Right, though surfacekinetics was assumed to be added last + `nb_surfaces` should be less or equal 2 in 1D

File: festim/exports/derived_quantities/adsorbed_hydrogen.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/exports/derived_quantities/adsorbed_hydrogen.py:7]: I think the description ("computes the hydrogen surface concentration on a surface") should be changed to something more explicit and more descriptive of the actual context it should be used with (surface kinetics)

File: festim/exports/derived_quantities/adsorbed_hydrogen.py (hunk 2 above)
Comment (KulaginVladimir) [reply festim/exports/derived_quantities/adsorbed_hydrogen.py:39]: Shall we change this with `self.function(0)`?

Comment (RemDelaporteMathurin) [reply festim/exports/derived_quantities/adsorbed_hydrogen.py:39]: We could change it to `self.function(x)` but then what is `x`? it can in theory be anything

I think that keeping this implementation is fine! we just have to raise an error if this is used with a non-1D simulation.

File: festim/exports/derived_quantities/adsorbed_hydrogen.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/exports/derived_quantities/adsorbed_hydrogen.py]: I updated it. Let me know if it needs more details.

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Another round of comments

File: festim/exports/derived_quantities/adsorbed_hydrogen.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/exports/derived_quantities/adsorbed_hydrogen.py]: perfetto 👌

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            test_function (fenics.TestFunction or ufl.Indexed): mobile test function
+            T (festim.Temperature): the temperature of the simulation
+            ds (fenics.Measure): the ds measure of the sim
+            dt (festim.Stepsize): the step-size
+        """
+
+        l_abs = self.l_abs
+        N_s = self.N_s
+        N_b = self.N_b
+        self.F = 0
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:91]: The `bc.form` of normal bcs differs a bit from what `SurfaceKinetics.F` stands for. If it's ok, then I'll re-name it.
On the another hand, `SurfaceKinetics.F` can be re-written to be more like other bc, but I don't think it'll simplify things, since both forms depend on mobile and adsorbed species.

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:91]: yes I saw that after writing my comment. I think this is fine

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+import sympy as sp
+
+
+class SurfaceKinetics(FluxBC):
+    """
+    FluxBC subclass allowing to include surface processes on in 1D H transport simulations.
+
+    d c_s / dt = k_bs l_abs c_m (1 - c_s/N_s) - k_sb c_s (1 - c_b/N_b) + J_vs
+
+    -D * grad(c) * n = -l_abs * d c / dt - k_bs l_abs c_m (1 - c_s/N_s) + k_sb c_s (1 - c_b/N_b)
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:12]: Sure, but I doubt on `l_abs`, which is not strictly defined in papers. For example, @ehodille [used](https://iopscience.iop.org/article/10.1088/1741-4326/ad2a29) different values for `l_abs` in these equations. In the second eq.: `l_abs` - the distance between two IS in W, whereas in the first one: `l_abs=N_s/N_b` - distance between subsurface IS and adsorption sites. As I remember, in some other works this characteristic distance was the same in both eqs. I assume that one value for `l_abs` can be used if `N_s=N_b^(2/3)`, `l_abs=N_b^(-1/3)`.

File: festim/h_transport_problem.py [hunk 8]
```diff
@@ -141,11 +152,34 @@ def initialise_concentrations(self):
...
+                if isinstance(concentration, festim.SurfaceKinetics):
+                    # iterate through each surface of each SurfaceKinetics
+                    for i in range(len(concentration.surfaces)):
+                        concentration.solutions[i] = self.u.sub(index)
+                        concentration.previous_solutions[i] = self.u_n.sub(index)
+                        concentration.test_functions[i] = list(split(self.v))[index]
+                        index += 1
+                else:
+                    concentration.solution = self.u.sub(index)
+                    # concentration.solution = list(split(self.u))[i]
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/h_transport_problem.py:179]: Comment?

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:12]: We need to clarify this and cite the paper from which this exact model is taken from. 

Another discrepancy with the paper from Etienne you cite is that the coverage term in `\varphi_{gas}` is squared and not in the current implementation

File: festim/h_transport_problem.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/h_transport_problem.py:179]: yes the comment. I think we can remove it

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:12]: i see `(1-c_b/N_b)` in your equation, what's `c_b`?

[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:12]: It's a typo. It should be `c` - solute concentration (I used earlier `c_b` where `b` standed for the bulk).

[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:12]: Can we just define the rate constant for the bulk-to-surface transition in `m/s`? That way, the ODE for the surface concentration would become: $\frac{dc_s}{dt}=k_{bs}c_m(1-\frac{c_s}{N_s})-k_{sb}c_s + J_{vs}$. For example:
```python
def k_bs(T, cs):
    nu=10e13
    lambda1 = 110e-12
    E_bs = 0.2
    return nu*lambda1*f.exp(-E_bs/F.k_B/T)
``` 

Thus, the users would have to provide `l_abs` (can be differ from `lambda1` for the H transport BC.

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:12]: I don't see why this would be an issue

**Comment by KulaginVladimir** [comment]
I noticed that `SurfaceKinetics` breaks the right content menu, so the subsection `Flux` doesn't open

image

**Comment by RemDelaporteMathurin** [comment]
> I noticed that `SurfaceKinetics` breaks the right content menu in API, so the subsection `Flux` doesn't open

Right this doesn't occur on main. This is a wild guess but could it be because of the warning block? Does this appear when you compile locally?

File: festim/generic_simulation.py [hunk 9]
```diff
@@ -349,6 +366,15 @@ def initialise(self):
                         warnings.warn(
                             f"{type(q)} may not work as intended for {self.mesh.type} meshes"
                         )
+                    if isinstance(q, festim.AdsorbedHydrogen) and not any(
+                        q.surface in bc.surfaces
+                        for bc in self.boundary_conditions
+                        if isinstance(bc, festim.SurfaceKinetics)
+                    ):
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/generic_simulation.py:373]: OK, but maybe we will use a nested if-statement, so we won't need to collect `SurfaceKinetics` bcs for each derived quantity? 

```python
if isinstance(q, festim.AdsorbedHydrogen):
    # check that festim.AdsorbedHydrogen is defined together with 
    # festim.SurfaceKinetics on the same surface
    surf_kin_present = any(
        q.surface in bc.surfaces
        for bc in self.boundary_conditions
        if isinstance(bc, festim.SurfaceKinetics)
    )
    
    if not surf_kin_present:
        raise AttributeError(
            f"SurfaceKinetics boundary condition must be defined on surface {q.surface} to export data with festim.AdsorbedHydrogen"
        )
```

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
+from festim import FluxBC
+from fenics import *
+import sympy as sp
+
+
+class SurfaceKinetics(FluxBC):
+    """
+    FluxBC subclass allowing to include surface processes on in 1D H transport simulations.
+
+    d c_s / dt = k_bs l_abs c_m (1 - c_s/N_s) - k_sb c_s (1 - c_b/N_b) + J_vs
```
Comment (KulaginVladimir) [inline festim/boundary_conditions/fluxes/surface_kinetics.py:10]: @RemDelaporteMathurin 
The issue with API appears even if I delete everything (Args, Attributes, ..) except text description. Also, the issue appears when I compile locally

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py]: So, I used a similar notation to Etienne's model, but kept the $(1-c_m/n_{IS})$ term. As long as $c_m << n_{IS}$, the models are equal. Otherwise, I couldn't reproduce the results from [this](https://www.sciencedirect.com/science/article/pii/S2352179120300272) paper on H absorption in Ti.

File: festim/generic_simulation.py [hunk 11]
```diff
@@ -226,32 +226,49 @@ def attribute_boundary_conditions(self):
...
 
         for bc in self.boundary_conditions:
             if bc.field not in valid_fields:
                 raise ValueError(f"{bc.field} is not a valid field for BC")
+            # check SurfaceKinetics in 1D simulations
+            if (
+                isinstance(bc, festim.SurfaceKinetics)
+                and self.mesh.mesh.topology().dim() != 1
+            ):
+                raise ValueError("SurfaceKinetics can only be used in 1D simulations")
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/generic_simulation.py:241]: I added the `check_boundary_conditions` method.

File: festim/generic_simulation.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/generic_simulation.py]: For now, I made as shown above

File: festim/h_transport_problem.py [hunk 12]
```diff
@@ -176,6 +210,20 @@ def initialise_concentrations(self):
...
                 )
 
+        # assign initial condition for SurfaceKinetics BC
+        # iterate through each surface of each SurfaceKinetics
+        index = len(self.traps) + 1
+        for bc in [
+            bc
+            for bc in self.boundary_conditions
+            if isinstance(bc, festim.SurfaceKinetics)
+        ]:
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply festim/h_transport_problem.py]: done. Let me know if it looks ok

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 10 above)
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py]: I found that changing the subsection title (e.g. [this](https://github.com/festim-dev/FESTIM/blob/b56a3dc1dca432109d489c012aefea280698b807/docs/source/api/boundary_conditions.rst?plain=1#L31-L32)) to a higher level solves the problem

Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py]: I still don't see the classes appear: 
image

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py]: Thanks! I'll close this comment to open a new one.

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+    .. math::
+
+        \dfrac{d c_{\mathrm{s}}}{dt} = k_{\mathrm{bs}} c_{\mathrm{m}} \lambda_{\mathrm{abs}} (1 - \dfrac{c_\mathrm{s}}{n_{\mathrm{surf}}})
+        - k_{\mathrm{sb}} c_{\mathrm{s}} (1 - \dfrac{c_{\mathrm{m}}}{n_\mathrm{IS}}) + J_{\mathrm{vs}};
+
+        -D \dfrac{\partial c_{\mathrm{m}}}{\partial x} = -\lambda_{\mathrm{IS}} \dfrac{\partial c_{\mathrm{m}}}{\partial t}
+        - k_{\mathrm{bs}} c_{\mathrm{m}} \lambda_{\mathrm{abs}} (1 - \dfrac{c_\mathrm{s}}{n_{\mathrm{surf}}})
+        + k_{\mathrm{sb}} c_{\mathrm{s}} (1 - \dfrac{c_{\mathrm{m}}}{n_\mathrm{IS}}),
+
+    where :math:`c_{\mathrm{m}}` is the concentration of solute hydrogen, :math:`c_{\mathrm{s}}` is the surface concentration of hydrogen
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/surface_kinetics.py:19]:
```suggestion
    where :math:`c_{\mathrm{m}}` is the concentration of mobile hydrogen (H/m3), :math:`c_{\mathrm{s}}` is the surface concentration of adsorbed hydrogen (H/m2)
```

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+    FluxBC subclass allowing to include surface processes in 1D H transport simulations
+
+    .. math::
+
+        \dfrac{d c_{\mathrm{s}}}{dt} = k_{\mathrm{bs}} c_{\mathrm{m}} \lambda_{\mathrm{abs}} (1 - \dfrac{c_\mathrm{s}}{n_{\mathrm{surf}}})
+        - k_{\mathrm{sb}} c_{\mathrm{s}} (1 - \dfrac{c_{\mathrm{m}}}{n_\mathrm{IS}}) + J_{\mathrm{vs}};
+
+        -D \dfrac{\partial c_{\mathrm{m}}}{\partial x} = -\lambda_{\mathrm{IS}} \dfrac{\partial c_{\mathrm{m}}}{\partial t}
+        - k_{\mathrm{bs}} c_{\mathrm{m}} \lambda_{\mathrm{abs}} (1 - \dfrac{c_\mathrm{s}}{n_{\mathrm{surf}}})
+        + k_{\mathrm{sb}} c_{\mathrm{s}} (1 - \dfrac{c_{\mathrm{m}}}{n_\mathrm{IS}}),
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/surface_kinetics.py:17]: This is very good! Could we split things up in fluxes like `J_{bs}`, `J_{sb}` etc? it will improve clarity imo

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+    where :math:`c_{\mathrm{m}}` is the concentration of solute hydrogen, :math:`c_{\mathrm{s}}` is the surface concentration of hydrogen
+    adsorbed on a surface, :math:`\lambda_{\mathrm{abs}}=n_{\mathrm{surf}}/n_{\mathrm{IS}}`
+
+    Reference: E.A. Hodille et al 2017 Nucl. Fusion 57 056002
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/surface_kinetics.py:22]: If the model isn't strictly equal to this paper maybe we can write instead of "reference", "For more details see:"

We could also add [this reference](https://www.sciencedirect.com/science/article/pii/S2352179120300272) too

File: festim/generic_simulation.py [hunk 16]
```diff
@@ -215,44 +215,66 @@ def attribute_source_terms(self):
             else:
                 field_to_object[source.field].sources.append(source)
 
-    def attribute_boundary_conditions(self):
-        """Assigns boundary_conditions to mobile and T"""
-        self.T.boundary_conditions = []
-        self.h_transport_problem.boundary_conditions = []
-
+    def check_boundary_conditions(self):
```
Comment (RemDelaporteMathurin) [inline festim/generic_simulation.py:218]:
```suggestion
    def check_boundary_conditions(self):
        """Runs a series of checks on the BCs and raise errors accordingly"""
```

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 10 above)
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py]: Right, I haven't figure out how to fix this. I kept the docstrings for `SurfaceKinetics` for now. Two things I noticed: if you fix `SievertsBC` then the Dirichlet classes also dissapear; if you delete several Flux classes fro API then `SurfaceKinetics` and other classes appear.

File: test/system/test_misc.py [hunk 17]
```diff
@@ -3,6 +3,7 @@
 import numpy as np
 import pytest
 import os
+import re
```
Comment (RemDelaporteMathurin) [inline test/system/test_misc.py:6]: I think we can remove this import

File: festim/exports/derived_quantities/adsorbed_hydrogen.py [hunk 18]
```diff
@@ -10,7 +10,7 @@ class AdsorbedHydrogen(SurfaceQuantity):
     .. warning::
 
         The AdsorbedHydrogen export can be used only if the SurfaceBoundary
```
Comment (RemDelaporteMathurin) [inline festim/exports/derived_quantities/adsorbed_hydrogen.py:12]:
```suggestion
        The ``AdsorbedHydrogen`` export can be used only if the ``SurfaceKinetics``
```

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
A few more comments

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 13 above)
Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:19]: If math will work in API, the units can be shown as :math:`\mathrm{H m}^{-3}`, otherwise - OK

Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:19]: `\mathrm{H \ m}^{-3}` then! $\mathrm{H \ m}^{-3}$

File: festim/boundary_conditions/fluxes/surface_kinetics.py [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+    """
+    FluxBC subclass allowing to include surface processes in 1D H transport simulations:
+
+    .. math::
+
+        \dfrac{d c_{\mathrm{s}}}{dt} = J_{\mathrm{bs}} - J_{\mathrm{sb}} + J_{\mathrm{vs}};
+
+    .. math::
+
+        -D \dfrac{\partial c_{\mathrm{m}}}{\partial x} = -\lambda_{\mathrm{IS}} \dfrac{\partial c_{\mathrm{m}}}{\partial t}
```
Comment (RemDelaporteMathurin) [inline festim/boundary_conditions/fluxes/surface_kinetics.py:16]: This wouldn't be true for the right hand side boundary. Let's use the generic nabla operator instead. What do you think?

```suggestion
        -D \nabla c_\mathrm{m} \cdot \mathbf{n} = -\lambda_{\mathrm{IS}} \dfrac{\partial c_{\mathrm{m}}}{\partial t}
```

Comment (KulaginVladimir) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:16]: \nabla command doesn't work

Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:16]: oh really? hm that's weird cause it works here: $\nabla$

Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py:16]: Oh yes because python interprets \n as a line break. Adding `r"""` before the docstrings sign fixes it:

image

File: festim/boundary_conditions/fluxes/surface_kinetics.py (hunk 10 above)
Comment (RemDelaporteMathurin) [reply festim/boundary_conditions/fluxes/surface_kinetics.py]: Worked on in #797

**Comment by RemDelaporteMathurin** [comment]
@KulaginVladimir this is very good thank you so much for this AWESOME contribution! This opens up so many possibilities with FESTIM! 🚀 

Would you mind attaching to this PR an example script for us to play with, like the Ti validation case or maybe the model from @ehodille ?

**Review by RemDelaporteMathurin — APPROVED** [review]
Thanks for this @KulaginVladimir ! We can either add a Theory section in this PR or do it in another one. Either is fine with me.

**Comment by KulaginVladimir** [comment]
@RemDelaporteMathurin, I can attach both scripts as there are some differences. Or I can make a separate repository with these cases.

> We can either add a Theory section in this PR or do it in another one. Either is fine with me.

I'd like to add theory and UG in a separate PR to avoid possible conflicts. Is it OK?

**Comment by RemDelaporteMathurin** [comment]
> I can attach both scripts as there are some differences. Or I can make a separate repository with these cases.

Whatever's easiest for you. 

> I'd like to add theory and UG in a separate PR to avoid possible conflicts. Is it OK?

No problem!

**Comment by RemDelaporteMathurin** [comment]
image

I managed to reproduce the figures! This is a slick implementation!
